packages/@aws-cdk/core/lib/private/refs.ts

```diff
@@ -193,10 +193,20 @@ function findAllReferences(root: IConstruct) {
 function createImportValue(reference: Reference): Intrinsic {
   const exportingStack = Stack.of(reference.target);
 
-  const importExpr = exportingStack.exportValue(reference);
-
-  // I happen to know this returns a Fn.importValue() which implements Intrinsic.
-  return Tokenization.reverseCompleteString(importExpr) as Intrinsic;
+  let importExpr: any;
+  try {
+    importExpr = exportingStack.exportValue(reference);
+    // I happen to know this returns a Fn.importValue() which implements Intrinsic.
+    return Tokenization.reverseCompleteString(importExpr) as Intrinsic;
+  } catch (e) {
```

Comment:
I'd rather we do an `if` here on the same kind of internal check that `exportValue()` would use to determine whether to throw or not, rather than catch the exception.

This could be any kind of exception, not just the one you're thinking of.

Reply:
:+1: cleaner than catching the exception twice.